lib/logwriter.c

```diff
@@ -1419,6 +1468,7 @@ _register_counters(LogWriter *self)
 
   }
   stats_unlock();
+  _register_aggregated_stats(self);
```

Comment:
Here you have a `lock, unlock, lock, unlock` sequence, would it be possible to reduce it to `lock, unlock` ?

Reply:
unfortunately we can't, because it two different lock (stats_lock and stats_aggregator_lock). The stats_aggregator_lock is using the stats_lock (at the registry). If we use first stats_lock then using the stats_aggregator_lock we get recursive lock :/ 
I'm not sure it is possible to use only stats_lock without stats_aggregator_lock (and we have assert at stats_register_aggregator_maximum to stats_aggregator_lock is locked)